src/entities/Basket/Basket.controller.ts

```diff
@@ -0,0 +1,32 @@
+import StatusCodes from 'http-status-codes';
+import { Request, Response } from 'express';
+import validateBasketReq from './Basket.validation';
+import Basket from './Basket.schema'
+
+const { BAD_REQUEST, CREATED, OK } = StatusCodes;
+
+export const getBasketById = async (req: Request, res: Response) => {
+    try {
+        const basket = await Basket
+            .findById(req.params.id)
+            .populate('createdByUserId', 'name')
+            .populate('targetUserID', 'name')
+        return res.status(OK).json(basket);
+    } catch (error) {
+        return res.status(BAD_REQUEST).send(error.message);
+    }
+}
```

Comment:
Please add checking if the user has the right to see the basket - he/she has to be either the offerer or the recipient (use JWT)

Reply:
Could you give a a hint how to do it with JWT?

Reply:
```suggestion
export const getBasketById = async (req: Request, res: Response) => {
  const user= await User.findById(req.user);
  if (!user) return res.status(BAD_REQUEST).send("The request is not sent by logged user");
    try {
        const basket = await Basket
            .findById(req.params.id)
            .populate('createdByUserId', 'name')
            .populate('targetUserID', 'name')
        if (!basket.createdByUserId.equals(user._id) && !basket.targetUserID.equals(user._id)) {
        return.status(FORBIDDEN).send("Basket does not belong to the user"}   
        return res.status(OK).json(basket);
    } catch (error) {
        return res.status(BAD_REQUEST).send(error.message);
    }
}
```
Something like that should work, although of course I am not sure if that's the best way to do it ;)

Reply:
DONE -> 40dbc6cdcaa41a21b521efd1c203871df584c7ca